modules/imaging_browser/php/imaging_browser.class.inc

```diff
@@ -284,117 +98,97 @@ END";
-            FROM files WHERE FileType='mnc' AND OutputType!='native'",
-            array()
+        // get list of scan types that should be present in the data table
+        // Get the intersection between all the scan types and those
+        // that are desired to go into imaging browser table, based on
+        // array values rather than keys (hence the array_flip), then flip
+        // the resulting array back to revert it to a key/value (i.e.
+        // acquisition protocol ID/scan type) combination.
+        $config             = \NDB_Config::singleton();
+        $toTable_scan_types = $config->getSetting('tblScanTypes');
+        $scan_id_types      = array_flip(
+            array_intersect_key(
+                array_flip($all_scan_types),
+                array_flip($toTable_scan_types)
+            )
         );
 
-        $this->fieldOptions = [
-                               'sites'         => $siteList,
-                               'projects'      => $list_of_projects,
-                               'visitQCStatus' => $visitQCStatus,
-                               'sequenceTypes' => $scan_types,
-                               'pendingNew'    => $pending,
-                               'configLabels'  => $scan_id_types,
-                              ];
+        return [
+                'sites'         => $siteList,
+                'projects'      => $list_of_projects,
+                'visitQCStatus' => $visitQCStatus,
+                'sequenceTypes' => $all_scan_types,
+                'pendingNew'    => $pending,
+                'configLabels'  => $scan_id_types,
+               ];
 
-        $this->tpl_data['backURL'] = $_SERVER['REQUEST_URI'];
     }
+
     /**
-     * Converts this menu filter to an array of the form
-     *
-     * Headers => (string array)
-     * Data => (array of arrays of row data)
+     * Gets the data source for this menu filter.
      *
-     * @note overloaded function
-     *         Overloading this method to create a list of sessionID that
-     *         will be used for the Navigation Links in  the viewSession
-     *         page.
+     * @return \LORIS\Data\Provisioner
+     */
+    public function getBaseDataProvisioner() : \LORIS\Data\Provisioner
+    {
+        return new ImagingBrowserRowProvisioner();
+    }
+
+
+    /**
+     * Return a data provisioner of the same type as BaseDataProvisioner, with
+     * additional LORIS filters applied.
      *
-     * @return array
+     * @return \LORIS\Data\Provisioner a provisioner with specific filters added
      */
-    function toArray()
+    public function getDataProvisionerWithFilters(): \LORIS\Data\Provisioner
     {
-        $data  = parent::toArray();
-        $index = array_search('SessionID', $data['Headers']);
-        if ($index !== false) {
-            $_SESSION['State']->setProperty(
-                'mriSessionsListed',
-                array_column($data['Data'], $index)
+
```

Comment:
It's getting closer but:

1. I think it should start with `$provisioner = parent::getDataProvisionerWithFilters` instead of duplicating the code from it, so that other filters added to the base (ie. projects eventually) automatically get applied
2. I don't think the logic for the if statements is correct. It looks like the way it's written unless you have *both* permissions anything not at your site gets filtered out.. but I'm not entirely sure what the logic is supposed to be for the permissions intersection either (since the logic in the code was always confusing.)
3. You may need to create new filter for the imaging browser instead of using the UserSiteMatch, because the filter you're trying to use is dependent on both CenterID *and* Entity Type at the same time, not just CenterID.

Reply:
1. oh right. Fixed in the next commit
2. oops... Yeah.. I spent a fair amount of time thinking about this logic. I think I finally got it! Check the next commit
3. some of the filters were already created by Xavier so I used them in the code. I just had to create an additional filter for users that have permission view_ownsite + phantom_allsites. For everything else, I just needed to call the right filter.

Thank you for your help and guidance!!